python/python_next/ir.cpp

```diff
@@ -38,10 +38,95 @@ void bindBaseNodes(py::module& nvfuser) {
       nvfuser, "Expr");
 }
 
+void bindInternalBaseNodes(py::module& nvfuser) {
+  // IterDomain
+  py::class_<IterDomain, Val, std::unique_ptr<IterDomain, py::nodelete>>(
+      nvfuser, "IterDomain")
+      .def(
+          "__str__",
+          [](IterDomain* self) { return self->toString(/*indent_size=*/0); },
+          "Convert the IterDomain to a string representation.")
+      .def(
+          "extent",
+          &IterDomain::extent,
+          R"(
+Get the extent of this domain.
+
+Returns
+-------
+Val
+    The extent of this domain.
+)");
+
+  // TensorDomain
+  py::class_<TensorDomain, Val, std::unique_ptr<TensorDomain, py::nodelete>>(
+      nvfuser, "TensorDomain")
+      .def(
+          "__str__",
+          [](TensorDomain* self) { return self->toString(/*indent_size=*/0); },
+          "Convert the TensorDomain to a string representation.");
+}
+
+void bindInterfaceNodes(py::module& nvfuser) {
+  py::class_<TensorView, Val, std::unique_ptr<TensorView, py::nodelete>>(
+      nvfuser, "TensorView")
+      .def(
+          "__str__",
+          [](TensorView* self) { return self->toString(/*indent_size=*/0); },
+          "Convert the TensorView to a string representation.")
+      .def(
+          "num_dims",
+          &TensorView::nDims,
+          R"(
+Get the number of dimensions in this tensor.
+
+Returns
+-------
+int
+    The number of dimensions.
+)")
+      .def(
+          "domain",
+          &TensorView::domain,
+          R"(
+Get the TensorDomain of this tensor.
+
+Returns
+-------
+TensorDomain
+    The tensor domain object that describes the dimensionality and properties
+    of this tensor. The tensor domain contains information about:
+    - Root domain (The original dimensions if logical domain contains rFactor iterDomains.)
+    - Logical domain (The original dimensions. It may contain rFactor iterDomains.)
+    - Allocation domain (How the memory is allocated for the tensor?)
+    - Loop domain (The for-loop structure for the tensor.)
+)")
+      .def(
+          "axis",
+          &TensorView::axis,
+          py::arg("index"),
+          py::return_value_policy::reference,
+          R"(
+Get the iteration domain at the specified axis.
+
+Parameters
+----------
+index : int
+    The axis index.
+
+Returns
+-------
+IterDomain
+    The iteration domain at the specified axis.
+)");
+}
+
 } // namespace
 
 void bindFusionIr(py::module& nvfuser) {
   bindBaseNodes(nvfuser);
+  bindInternalBaseNodes(nvfuser);
+  bindInterfaceNodes(nvfuser);
```

Comment:
Let's leave it as-is for now.

Given the amount of comments, I think we'll inevitably want to split `InternalBaseNodes` and `InterfaceNode` into separate files at a later point.